workflow/controller/workflowpod.go

```diff
@@ -151,6 +151,9 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 	wfSpec := woc.execWf.Spec.DeepCopy()
 
 	mainCtr.Name = common.MainContainerName
+	if isResourcesSpecified(woc.controller.Config.MainContainer) {
+		mainCtr.Resources = woc.controller.Config.MainContainer.Resources
```

Comment:
1. The workflow spec should take priority over the config.
2. Test?

Reply:
Thanks for the feedback. Just updated the PR.

Reply:
I think you can ignore `podSpePatch`, but you should check tmpl.Container

Reply:
Could you clarify why we need to check tmpl.Container? The goal of this PR is to customize main container no matter what's in the template (similar to what's currently done for executor).

Reply:
that would be contrary to how we do it for other item - the user should be able to override a global setting like this

Reply:
Actually another goal of this PR is to set reasonable default resources for the main container.

I've added a check on `tmpl.Container`. Let me know if I've missed anything here.